### PR TITLE
Test engine version hash, but skip for Dart HHH bot

### PR DESCRIPTION
**Pull request by dnfield**
Relands #55057 with a special condition to skip if we're on Dart's LUCI infra, where they do strange things to the cache.

I think this skip condition is not ideal, but it is better than just disabling the test permanently and should allow for time for Dart infra to implement an alternative if one is possible.

/cc @alexmarkov @athomas

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -116,8 +130,37 @@ Future<void> main(List<String> args) async {
...
+  final String luciBotId = Platform.environment['SWARMING_BOT_ID'] ?? '';
+  if (luciBotId.startsWith('luci-dart-')) {
+    // The Dart HHH bots intentionally modify the local artifact cache
+    // and then use this script to run Flutter's test suites.
+    // Because the artifacts have been changed, this particular test will return
+    // a false positive and should be skipped.
+    print('${yellow}Skipping Flutter Engine Version Validation for swarming '
+          'bot $luciBotId.');
+    return;
+  }
```
Comment (dnfield) [inline dev/bots/test.dart:145]: This is the new part from last time (minor some slight refactoring).  Bot ID was taken from https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket.appspot.com/8882487817844098672/+/steps/flutter_test_add_to_app_tests/0/logs/execution_details/0

File: dev/bots/test.dart [hunk 2]
```diff
@@ -116,8 +130,37 @@ Future<void> main(List<String> args) async {
   print('$clock ${bold}Test successful.$reset');
 }
 
+/// Verify the Flutter Engine is the revision inin
```
Comment (digiter) [inline dev/bots/test.dart:133]: "inin" should be "in"

**Review by digiter — APPROVED** [review]
LGTM

**Comment by jonahwilliams** [comment]
Can the HHH bot stop doing strange things to the cache? I would be worried about any cache verification updates we do in the future breaking. FYI @christopherfujino

**Comment by christopherfujino** [comment]
For reference, where is the HHH recipe? (I always struggle discovering LUCI recipes...) @athomas ?

File: dev/bots/test.dart (hunk 2 above)
Comment (dnfield) [reply dev/bots/test.dart:133]: Fixed

**Comment by dnfield** [comment]
Filed https://github.com/dart-lang/sdk/issues/41588 to request changes to Dart infra, with a link to the recipe that drives the HHH bot.

**Pull request merged**